pkg/event/schema/introspectRequest.message.yaml

```diff
@@ -0,0 +1,36 @@
+# Message schema for introspectRepository
+$schema: "http://json-schema.org/draft-07/schema#"
+$id: "https://github.com/content-services/content-sources-backend/api/introspectRepository.message.yaml"
+# $ref: "#/definitions/Introspect"
+title: Kafka Message Schemas
+description: Schema for the introspect repository kafka message
+type: object
+additionalProperties: true
+properties:
+  uuid:
+    description: |
+      The UUID for the repository to be introspected. This
+      is used for the key field to distribute the messages
+      to the consumers.
+    type: string
+    format: uuid
+  url:
+    description: The base url for the repository to be introspected
+    type: string
+    # TODO Pattern is not implemented for the code generator leveraged
+    #      pattern field is according to the ECMA-262 regular expression dialect
+    #      (https://262.ecma-international.org/13.0/#sec-regexp-regular-expression-objects)
+    #      however golang use the syntax accepted by RE2 (https://pkg.go.dev/regexp
+    #      https://github.com/google/re2/wiki/Syntax)
+    #      The json schema documentation recommend to use only a ubset of the regexp:
+    #      https://json-schema.org/understanding-json-schema/reference/regular_expressions.html
+    # TODO add format: regex and the pattern to be matched
+    # format: regex
+    # pattern: |
+    #   `^(?P<origin>(?P<protocol>http[s]?:)?\/\/(?P<host>[a-z0-9A-Z-_.]+))(?P<port>:\d+)?(?P<path>[\/a-zA-Z0-9-\.]+)?(?P<search>\?[^#\n]+)?(?P<hash>#.*)?$`gm
+    # https://regex101.com/r/kE6aO8/1
+    # pattern: |
+    #   /^(?:(?:https?|ftp):\/\/)(?:\S+(?::\S*)?@)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)(?:\.(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,}))\.?)(?::\d{2,5})?(?:[/?#]\S*)?$/i
+required:
+  - uuid
+  - url
```

Comment:
why is there a json and yaml version of the same thing?  (or is there some difference?)

Reply:
I try to clarify below:

- If I remove the `.json` files, the tool yaml2json will be required in container build time, and building time (which will impact into the pipeline too), requiring the dependency to be installed.
- why is this happening? `make build` has a dependency on `make gen-event-messages`; this second rule has a general rule that generate the json version, if it does not exist, or if the yaml version is updated; to do that it uses yaml2json tool, which is a python tool.
- the json version is embedded into the golang code.
- the json version is required because the schema is read in json format by the library `https://github.com/qri-io/jsonschema`.

Because of all the above I left the json version into the repository even when it can be re-generated; it is a sequence of linked consequences :face_with_spiral_eyes:

I have to review `README.md`, and if the dependency for development on `yaml2json` is not mentioned, I have to add it there; this dependency is the same to use json2yaml

Reply:
update: I have added the requirement to `README.md` file.